### DISCUSSION TITLE
API to interrogate model/chat

### DISCUSSION THREAD

**Discussion by SimoneFaricelli** [open-webui/open-webui#3300] [General]
**Describe the solution you'd like**
I'd like to have an API to call to interrogate a model I created through the interface. Is there any way? I checked all the APIs on the path /api/v1/docs, but i couldn't find any.

**Comment by AndreiKozlov** [comment]
Example code PYTHON
`
import requests

# URL and API
api_url = "http://localhost:8080/ollama/api/generate"
api_key = "PAST YOUR API KEY"

# Heeaders Authorizations
headers = {
    "Authorization": f"Bearer {api_key}",
    "Content-Type": "application/json"
}

# example request - model llama3
data = {
    "model": "llama3",
    "prompt": "Why is the sky blue?",
    "stream": 'false'
}

# Send Post requests
response = requests.post(f"{api_url}", headers=headers, json=data)

# response
if response.status_code == 200:
    response_data = response.json()
    print("RESPONSE:", response_data)
else:
    print("ERROR:", response.status_code, response.text)
`

**Reply by SimoneFaricelli** [reply]
Thanks for the reply, I didn't explain it well. I want to make inference on (to chat with) the assistants created through OpenWeb UI interface, not the model itself. Is it duable?

**Comment by SimoneFaricelli** [comment]
Any Help? I'd like to emulate the behaviour of OpenAI assistants.
